src/test/java/core/basesyntax/service/impl/WriterServiceImplTest.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.ReportCreatorService;
+import core.basesyntax.service.WriterService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.Before;
+import org.junit.Test;
+
+public class WriterServiceImplTest {
+    private WriterService writerService;
+    private ReportCreatorService reportCreatorService;
+
+    @Before
+    public void setUp() throws Exception {
+        writerService = new WriterServiceImpl();
+        reportCreatorService = new ReportCreatorServiceImpl();
+    }
+
+    @Test
+    public void writeToFile_OK() {
+        String path = "src/main/resources/report.csv";
+        String expected = null;
+        try {
+            expected = Files.readString(Path.of("src/main/resources/report.csv"));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        String actual = reportCreatorService.createReport();
+        writerService.writeToFile(path, actual);
+        assertEquals(expected, actual);
```

Comment:
let's have a report as string
then write it to file
then read this file and check if its content and your report are the same